subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c

```diff
@@ -139,6 +149,29 @@ static void cu_update_conn_parameters(struct ll_conn *conn, struct proc_ctx *ctx
 				   ctx->data.cu.instant);
 }
 
+#if defined(CONFIG_BT_CTLR_CONN_PARAM_REQ)
+static bool cu_check_conn_parameters(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	const uint16_t interval_min = ctx->data.cu.interval_min;
+	const uint16_t interval_max = ctx->data.cu.interval_max; /* unit conn events (ie 1.25ms) */
+	const uint16_t timeout = ctx->data.cu.timeout; /* unit 10ms */
+	const uint16_t latency = ctx->data.cu.latency;
+	const uint16_t preferred_periodicity = ctx->data.cu.preferred_periodicity;
+
+	/* Invalid parameters */
+	const bool invalid = ((interval_min < CONN_INTERVAL_MIN(conn)) ||
+	    (interval_max > 3200) ||
+	    (interval_min > interval_max) ||
+	    (latency > 499) ||
+	    (timeout < 10) || (timeout > 3200) ||
+	    ((timeout * 4U) <= /* *4U re. conn events is equivalent to *2U re. ms */
+	     ((latency + 1) * interval_max)) ||
+	    (preferred_periodicity > interval_max));
```

Comment:
Minor: I know you just ported the old code, but maybe we should come up with some constants for these magic numbers